### ISSUE TITLE
scrollBehavior gets called two times on initial page load

### ISSUE THREAD

**Issue by antony-k1208** [nuxt/framework#9230]
### Environment

Nuxi 3.0.0                                                                                                                                                                             

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.15.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.5.5`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-drvjtv?file=app%2Frouter.options.ts,app.vue

### Describe the bug

When using `app/router.options.ts` to set a `scrollBehavior` callback for example to restore the old scroll position or scroll to an anchor it gets called two times on initial page load. 

Check the console output for `from` and `to` route information. It seems that the layout system is somehow messing with the routing as you can see in `to.meta` for the first and second call. Also `from.name`, `from.path` and `from.fullPath` are different.

Let me know if I can be of further help.

**Comment by danielroe** [comment]
I think this is expected. We initialise the router, then 'replace' the same route to it to enable middleware to run on client-side initial load.

**Comment by antony-k1208** [comment]
> 

Thanks for your incredibly quick response 🤓. I see, I see. What's difficult with that is, that both calls have different params, especially when it comes to the savedPosition argument.

From what I saw is the first one the correct one, so with the corresponding logic in place you would scroll to the right position but as there is the second call with no savedPosition following along you immediately jump back up again.

**Comment by danielroe** [comment]
Do you think the issue is just the `savedPosition`?

**Comment by antony-k1208** [comment]
Yes, I think so. That's the whole point of using `scrollBehavior`, I guess. And I really don't know if it's really that big of a deal. I just discovered it while developing and wanted to let you guys know =)
